fe/fe-core/src/main/java/com/starrocks/sql/common/SyncPartitionUtils.java

```diff
@@ -301,20 +309,25 @@ public static Range<PartitionKey> transferRange(Range<PartitionKey> baseRange,
             truncUpperDateTime = upperDate.toLocalDateTime();
         } else {
             upperDate = (DateLiteral) upperExpr;
-            truncUpperDateTime = getLowerDateTime(upperDate.toLocalDateTime(), granularity);
+            truncUpperDateTime = getUpperDateTime(upperDate.toLocalDateTime(), granularity);
         }
 
         Preconditions.checkState(baseRange.lowerEndpoint().getTypes().size() == 1);
         PrimitiveType partitionType = baseRange.lowerEndpoint().getTypes().get(0);
 
         PartitionKey lowerPartitionKey = new PartitionKey();
         PartitionKey upperPartitionKey = new PartitionKey();
-        if (partitionType == PrimitiveType.DATE) {
-            lowerPartitionKey.pushColumn(new DateLiteral(truncLowerDateTime, Type.DATE), partitionType);
-            upperPartitionKey.pushColumn(new DateLiteral(truncUpperDateTime, Type.DATE), partitionType);
-        } else {
-            lowerPartitionKey.pushColumn(new DateLiteral(truncLowerDateTime, Type.DATETIME), partitionType);
-            upperPartitionKey.pushColumn(new DateLiteral(truncUpperDateTime, Type.DATETIME), partitionType);
+        try {
+            if (partitionType == PrimitiveType.DATE) {
+                lowerPartitionKey.pushColumn(new DateLiteral(truncLowerDateTime, Type.DATE), partitionType);
+                upperPartitionKey.pushColumn(new DateLiteral(truncUpperDateTime, Type.DATE), partitionType);
+            } else {
+                lowerPartitionKey.pushColumn(new DateLiteral(truncLowerDateTime, Type.DATETIME), partitionType);
+                upperPartitionKey.pushColumn(new DateLiteral(truncUpperDateTime, Type.DATETIME), partitionType);
+            }
+        } catch (AnalysisException e) {
+            throw new SemanticException("Convert partition with date_trunc expression to date failed, lower:%s, upper:%s",
```

Comment:
```suggestion
            throw new SemanticException("Convert partition with date_trunc expression to date failed, lower:%s, upper:%s",
```
should we add e into new SemanticException?

Reply:
no need? AnalysisException is thrown when `DateLiteral` is invalid, all infos are contained in the error message:
```
       if (isNullable()) {
            throw new AnalysisException("Invalid date value");
        }
```